File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/PurchaceOrderDto.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using ProjectPos.Data.Shared.Enums;
using ProjectPos.Services.EntityDtos;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectPos.Services.DTOs
{
    public class PurchaceOrderDto : FullAuditedEntityDto<int>
    {
        [Column(TypeName = "double(12, 2)")]
        public double? InvoiceValue { get; set; }
        public string? SupplierName { get; set; }
        public bool IsReceived { get; set; }
        public bool IsOpen { get; set; }
        public int? SupplierId { get; set; }
        public bool IsApproved { get; set; }
        public int? ApprovedById { get; set; }
        public DateTime ETA { get; set; }
        public OrderReceivedStatus? Status { get; set; }

        //Foreign Keys
        [ForeignKey("SupplierId")]
        public CompanyDto? Company { get; set; }

        //dependent entities
        public List<PurchaceOrderLineDto>? PurchaceOrderItems { get; set; }
    }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/purchace-order-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { OrderReceivedStatus } from "../enums/order-received-status";
import { CompanyDto } from "./company-dto";
import { GoodsReceivedVoucherDto } from "./goods-received-voucher-dto";
import { PurchaceOrderLineDto } from "./purchace-order-line-dto";

export interface PurchaceOrderDto extends FullAuditedEntityDto<number> {
    invoiceValue?: number;
    isReceived: boolean;
    isOpen: boolean;
    supplierId?: number;
    supplierName?: string;
    isApproved: boolean;
    approvedById?: number;
    eta: Date;
    status?: OrderReceivedStatus;
    company?: CompanyDto;
    purchaceOrderItems?: PurchaceOrderLineDto[];
    goodsReceivedVouchers?: GoodsReceivedVoucherDto[];
}
